### PR TITLE
Send feedback Form To external coordinating poc

### PR DESCRIPTION
**Pull request by tarunnjoshi**
Send feedback Form To external coordinating poc
Requirement: https://github.com/coloredcow-admin/goonj-crm/issues/170#issuecomment-2556349735

## Summary by CodeRabbit

- **New Features**
	- Enhanced visit status management with automatic email notifications for completed visits.
	- Introduced methods for validating visit statuses and generating feedback email content.
- **Bug Fixes**
	- Improved error handling for missing email addresses during notifications.

### CODE REVIEW

File: wp-content/civi-extensions/goonjcustom/Civi/UrbanPlannedVisitService.php [hunk 1]
```diff
@@ -83,4 +85,136 @@ private static function getOutcomeEmailHtml($coordinatingGoonjPOCName, $visitId)
...
+
+    if (!$visitStatusDetails) {
+      return;
+    }
+
+    $newVisitStatus = $visitStatusDetails['newVisitStatus'];
+    $currentVisitStatus = $visitStatusDetails['currentVisitStatus'];
+
+    if ($currentVisitStatus !== $newVisitStatus) {
+      if ($newVisitStatus === 'completed') {
```
Comment (belwalshubham) [inline wp-content/civi-extensions/goonjcustom/Civi/UrbanPlannedVisitService.php:111]: can you combine this to only one if ?

File: wp-content/civi-extensions/goonjcustom/Civi/UrbanPlannedVisitService.php [hunk 2]
```diff
@@ -83,4 +85,136 @@ private static function getOutcomeEmailHtml($coordinatingGoonjPOCName, $visitId)
...
+        $externalCoordinatingPocId = $objectRef['Urban_Planned_Visit.External_Coordinating_PoC'] ?? '';
+
+        $externalCoordinatingGoonjPOC = Contact::get(FALSE)
+          ->addSelect('email.email', 'display_name')
+          ->addJoin('Email AS email', 'LEFT')
+          ->addWhere('id', '=', $externalCoordinatingPocId)
+          ->execute()->single();
+
+        $externalCoordinatingGoonjPOCEmail = $externalCoordinatingGoonjPOC['email.email'];
+        $externalCoordinatingGoonjPOCName = $externalCoordinatingGoonjPOC['display_name'];
```
Comment (belwalshubham) [inline wp-content/civi-extensions/goonjcustom/Civi/UrbanPlannedVisitService.php:137]: can you fix the naming here ?
changed this `externalCoordinatingGoonjPOCEmail`
to `externalCoordinatingGoonjPocEmail`

and Please check other places here

**Review by belwalshubham — APPROVED** [review]

File: wp-content/civi-extensions/goonjcustom/Civi/UrbanPlannedVisitService.php (hunk 1 above)
Comment (tarunnjoshi) [reply wp-content/civi-extensions/goonjcustom/Civi/UrbanPlannedVisitService.php]: sure

File: wp-content/civi-extensions/goonjcustom/Civi/UrbanPlannedVisitService.php (hunk 2 above)
Comment (tarunnjoshi) [reply wp-content/civi-extensions/goonjcustom/Civi/UrbanPlannedVisitService.php]: sure

**Pull request merged**